core/src/accounts_hash_verifier.rs

```diff
@@ -147,6 +147,30 @@ impl AccountsHashVerifier {
             )
             .unwrap();
 
+        assert_eq!(accounts_package.expected_capitalization, lamports);
+        if let Some(expected_hash) = accounts_package.accounts_hash_for_testing {
+            // before we assert, run the hash calc again. This helps track down whether it could have been a failure in a race condition possibly with shrink.
+            // We could add diagnostics to the hash calc here to produce a per bin cap or something to help narrow down how many pubkeys are different.
+            if expected_hash != accounts_hash {
+                let _ = accounts_package
+                    .accounts
+                    .accounts_db
+                    .calculate_accounts_hash_without_index(
+                        &CalcAccountsHashConfig {
+                            use_bg_thread_pool: false,
```

Comment:
For completeness, we had this as `true` on the first pass. Really really really shouldn't affect anything tho

Reply:
yes, it was true on the first pass. But, with this pass, we are only trying to assert asap but leave behind some hopefully useful info. So, I chose to run it in the foreground with more threads.